### PR TITLE
Refactor, add specific command flags

### PR DESCRIPTION
**Pull request by 0xmaayan**
Will be released as a major bump
- Migrate to TS
- Introduce `--update` and `--install` flag commands
- Refactor code to smaller components
- Tool is no longer auto update the CLI when running it. If want to update should use the `--update` flag

### CODE REVIEW

**Review by 0xaptosj — APPROVED** [review]

File: bin/tasks/update.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import { execSyncShell } from "../utils/execSyncShell.js";
+import { getLatestVersion } from "../utils/getAptosCliLatestVersion.js";
+import { installCli } from "./install.js";
+import { getLocalBinPath } from "../utils/getLocalBinPath.js";
+
+export const updateCli = async () => {
+  const path = getLocalBinPath();
+  if (!existsSync(path)) {
+    console.log(
+      "Aptos CLI not installed, run `npx aptos --install` to install"
```
[the remark this answers is not in the document]
Comment (0xmaayan) [reply bin/tasks/update.ts:12]: by default, it `installs` if not installed and `runs` the CLI. The `--install` flag only installs (not runs) and `--update` flag updates (not runs)
https://github.com/aptos-labs/aptos-cli/pull/18/files#diff-e61b90e2501c1f18daa89d4e572db21435bb0c491cf9d9c20754f7f68ff1a12aR12-R28

File: package.json [hunk 2]
```diff
@@ -3,8 +3,35 @@
   "version": "0.2.0",
   "description": "Aptos CLI available from npmjs",
   "bin": {
-    "aptos": "bin/aptos"
+    "aptos": "./dist/aptos.js"
   },
-  "author": "",
-  "license": "ISC"
+  "type": "module",
+  "author": "aptoslabs.com",
```
Comment (gregnazario) [inline package.json:9]: Minor, probably just put `Aptos Labs`

**Review by gregnazario — APPROVED** [review]
Seems fine to me, if we can, we should add running a single command (such as an account lookup, or a init) to ensure the installation worked correctly for the github workflows.  I think it shows it only installing, but I could be wrong.

**Comment by 0xmaayan** [comment]
> Seems fine to me, if we can, we should add running a single command (such as an account lookup, or a init) to ensure the installation worked correctly for the github workflows. I think it shows it only installing, but I could be wrong.

Done. I had a `return` after installation so it didnt run. https://github.com/aptos-labs/aptos-cli/actions/runs/11351888400/job/31573452688?pr=18

**Pull request merged**